apps/studio/components/interfaces/SQLEditor/SQLEditor.tsx

```diff
@@ -296,9 +300,13 @@ const SQLEditor = () => {
           ? (selectedValue || editorRef.current?.getValue()) ?? snippet.snippet.content.sql
           : selectedValue || editorRef.current?.getValue()
 
-        const containsDestructiveOperations = checkDestructiveQuery(sql)
+        const destructiveOperations = checkDestructiveQuery(sql)
+        const updateWithoutWhereClause = isUpdateWithoutWhere(sql)
+
+        setContainsDestructiveOperations(destructiveOperations)
+        setContainsUpdateWithoutWhereClause(updateWithoutWhereClause)
 
-        if (!force && containsDestructiveOperations) {
+        if (!force && (containsDestructiveOperations || containsUpdateWithoutWhereClause)) {
```

Comment:
```suggestion
        if (!force && (destructiveOperations || updateWithoutWhereClause)) {
```

otherwise these will contain the stale values (set is asynchronous) and the modal will be bypassed

Reply:
yes good catch!